aten/src/THC/THCApply.cuh

```diff
@@ -165,13 +165,15 @@ inline dim3 getApplyBlock() {
   return dim3(THC_APPLY_THREADS_PER_BLOCK);
 }
 
-inline bool getApplyGrid(THCState* state, uint64_t totalElements, dim3& grid) {
-  int curDevice = -1;
-  cudaGetDevice(&curDevice);
-  if (curDevice == -1) return false;
+inline bool getApplyGrid(THCState* state, uint64_t totalElements, dim3& grid, int curDevice = -1) {
```

Comment:
Does this argument really needs to be optional? It's always specified

Reply:
Yes, there are a few places where getApplyGrid is called that I did not patch (FusedRNNCell and may be something else)